java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -411,6 +411,341 @@ public boolean isClosed() {
+
+        HttpResponse response = httpHelper.executeAndCloseRequest(put,
+                HttpStatus.SC_NO_CONTENT,
+                "PUT    {} response [{}] {} ");
+
+        // When LastModified is set, the bucket already exists
+        return response.getFirstHeader(HttpHeaders.LAST_MODIFIED) == null;
+    }
+
+    /**
+     * Deletes a bucket in Manta with the given path.
+     * @param rawPath Path of the bucket you want to delete.
+     * @param requestHeaders requestHeaders HTTP headers to attach to request (may be null)
+     * @throws IOException Problem getting object over the network
+     */
+    void deleteBucket(final String rawPath, final MantaHttpHeaders requestHeaders)
+            throws IOException {
+        Validate.notBlank(rawPath, "rawPath must not be blank");
+        final String path = formatPath(rawPath);
+        LOG.debug("DELETE {} bucket", path);
+
+        try {
+            httpHelper.httpDelete(path, requestHeaders);
+        } catch (MantaClientHttpResponseException e) {
+
+            // Attempting to delete a non-empty bucket will result in an error
+            if (e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_FOUND_ERROR)
+                    || e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_EMPTY_ERROR)) {
+                final MantaIOException mioe = new MantaIOException("Unable to delete bucket", e);
+                mioe.setContextValue("bucket", path);
+                throw mioe;
+            }
+        } catch (ConnectionPoolTimeoutException e) {
+            final MantaClientException clientException = new MantaClientException(
+                    "Connection pool timeout while deleting bucket" + e.getMessage());
+            clientException.setContextValue("path", path);
+            throw clientException;
+
+        } catch (IOException e) {
+            throw new IOException(e);
+        }
+        LOG.debug("Finished deleting bucket {}.", path);
+    }
+
+    /**
+     * Deletes a bucket from Manta.
+     *
+     * @param rawPath The fully qualified path of the Manta bucket.
+     * @throws IOException If an IO exception has occurred.
+     * @throws MantaClientHttpResponseException If a HTTP status code other than {@code 200 | 202 | 204} is encountered
+     */
+    public void deleteBucket(final String rawPath) throws IOException {
+        this.deleteBucket(rawPath, null);
+    }
+
+    /**
+     * <p>Finds all bucket objects under a given path. Since
+     * this method returns a {@link Stream}, consumers can add their own
+     * additional filtering based on object type or other criteria.</p>
+     *
+     * <p>Parallelism settings are set by JDK system property:
+     * <code>java.util.concurrent.ForkJoinPool.common.parallelism</code></p>
+     *
+     * <p><strong>WARNING:</strong> this method is not atomic and thereby not
+     * safe if other operations are performed on the bucket path while
+     * it is running.</p>
```

Comment:
I think this warning could be more direct regarding the consequences. Is this unsafe as in inconsistent (objects inserted concurrently may or may not show up in the list), or unsafe as in your objects might be lost?